odbc/src/api/functions.rs

```diff
@@ -901,10 +926,30 @@ pub unsafe extern "C" fn SQLEndTran(
 #[no_mangle]
 pub unsafe extern "C" fn SQLExecDirect(
     statement_handle: HStmt,
-    _statement_text: *const Char,
-    _text_length: Integer,
+    statement_text: *const Char,
+    text_length: Integer,
 ) -> SqlReturn {
-    unsupported_function(MongoHandleRef::from(statement_handle), "SQLExecDirect")
+    let query = input_text_to_string(statement_text, text_length as usize);
+    let mongo_handle = MongoHandleRef::from(statement_handle);
+    let stmt = must_be_valid!(mongo_handle.as_statement());
+    let mongo_statement = {
+        let stmt_guard = stmt.read().unwrap();
+        let connection = must_be_valid!((*stmt_guard.connection).as_connection());
+        let connection_guard = connection.read().unwrap();
+        let timeout = connection_guard.attributes.connection_timeout;
+        if let Some(ref mongo_connection) = connection_guard.mongo_connection {
+            MongoQuery::execute(mongo_connection, timeout, &query).map_err(|e| e.into())
+        } else {
+            Err(ODBCError::InvalidCursorState)
```

Comment:
I'm not sure I understand why the `mongo_connection` being `None` leads to this particular error. From the docs, I don't see how we can determine we are in this "invalid cursor state" based just on this information. Can you explain this?

Reply:
I can't find a better error.

Reply:
"general error"? 😂

I'm not convinced this case cleanly falls under any category on the docs page. That's surprising since it is such a long detailed list of error conditions. _Maybe_ "general error" would be better since in this case there isn't an open cursor since no `mongo_connection` even exists and "invalid cursor" errors are related to cursors being in specific states?